dist/bin/kctf-challenge

```diff
@@ -352,47 +352,92 @@ function kctf_chal_debug {
 }
 
 function kctf_chal_create_usage {
-  echo "usage: kctf chal create name" >&2
+  echo "usage: kctf chal create [args] name" >&2
+  echo "args:" >&2
+  echo "  -h|--help       print this help" >&2
+  echo "  --challenge-dir path where to create the new challenge" >&2
+  echo "                  default: \"${KCTF_CTF_DIR}/\${CHALLENGE_NAME}\"" >&2
 }
 
 function kctf_chal_create {
-  if [[ $# -ne 1 ]]; then
-    echo "kctf chal create: name missing" >&2
+  OPTS="h"
+  LONGOPTS="help,challenge-dir:"
+  PARSED=$(getopt --options=$OPTS --longoptions=$LONGOPTS --name "kctf chal create" -- "$@")
+  if [[ $? -ne 0 ]]; then
     kctf_chal_create_usage
     exit 1
   fi
+  eval set -- "$PARSED"
+
+  CHALLENGE_DIR=
+  while true; do
+    case "$1" in
+      -h|--help)
+        kctf_chal_create_usage
+        exit 0
+        ;;
+      --challenge-dir)
+        CHALLENGE_DIR="$2"
+        shift 2
+        ;;
+      --)
+        shift
+        break
+        ;;
+      *)
+        echo "Unrecognized argument \"$1\"." >&2
+        parse_help_arg_only_usage
+        exit 1
+        ;;
+    esac
+  done
 
-  if [[ "$1" == "-h" ]] || [[ "$1" == "--help" ]]; then
+  if [[ $# -ne 1 ]]; then
+    echo "kctf chal create: name missing" >&2
     kctf_chal_create_usage
-    exit 0
+    exit 1
   fi
 
   CHALLENGE_NAME="$1"
   shift
 
-  #we don't require anymore that the path is under the ctf directory.
-  #TODO: add a flag to control this
-  CHALLENGE_DIR="${KCTF_CTF_DIR}/${CHALLENGE_NAME}"
+  if [[ -z "${CHALLENGE_DIR}" ]]; then
+    CHALLENGE_DIR="${KCTF_CTF_DIR}/${CHALLENGE_NAME}"
+  else
+    CHALLENGE_DIR_REALPATH=$(realpath --canonicalize-missing "${CHALLENGE_DIR}")
+    if [[ "${CHALLENGE_DIR_REALPATH}" != "${KCTF_CTF_DIR}"/* ]]; then
```

Comment:
why is this necessary?

Reply:
we want to rely on this for versioning. I.e. we want to rely on the fact that every challenge is bound to the kctf version.
That's why I think it's good to (soft) enforce that challenges need to be under the CTF directory.

Reply:
k